src/control/cmd/dmg/pretty/pool.go

```diff
@@ -195,3 +197,79 @@ func PrintPoolProperties(poolID string, out io.Writer, properties ...*daos.PoolP
 	tf.InitWriter(out)
 	tf.Format(table)
 }
+
+// PrintPoolRanksResps generates a table showing results of operations on pool ranks. Each row will
+// indicate a common result for a group of ranks on a pool.
+func PrintPoolRanksResps(out io.Writer, resps ...*control.PoolRanksResp) error {
+	if len(resps) == 0 {
+		fmt.Fprintln(out, "No pool ranks processed")
+		return nil
+	}
+
+	// Results are aggregated based on error messages for a given pool ID.
+	poolErrRanks := make(map[string]map[string][]ranklist.Rank)
+	poolIDs := make(common.StringSet)
+	errMsgs := make(common.StringSet)
+
+	for _, resp := range resps {
+		if len(resp.Results) == 0 {
+			continue
+		}
+
+		id := resp.ID
+		poolIDs.Add(id)
+		if _, exists := poolErrRanks[id]; exists {
+			return errors.Errorf("multiple PoolRanksResps for the same pool %q", id)
+		}
+
+		seenRanks := make(map[ranklist.Rank]struct{})
+		poolErrRanks[id] = make(map[string][]ranklist.Rank)
+
+		for _, res := range resp.Results {
+			msg := "" // Key used for successful ranks,
+			if res.Errored {
+				msg = res.Msg
+			}
+
+			if _, exists := seenRanks[res.Rank]; exists {
+				return errors.Errorf("multiple PoolRankResults for rank %d", res.Rank)
+			}
```

Comment:
This is a type of case where I feel like returning an error is incorrect. If it's duplicated it could be logged and skipped (instead of ensuring the user doesn't get to see the rest of the command output).